.github/workflows/e2e-spin-sleep.yml

```
@@ -0,0 +1,184 @@
+name: Spin-Sleep End-to-End Tests
+
+on:
+  workflow_dispatch:
+    schedule:
+      - cron: "0 9 * * 1"
+  push:
+    branches: [main]
+    paths:
+      - "benchmarks/sleeping/**"
+      - "utils/**"
+      - "tools/**"
+      - "runner/**"
+
+  pull_request:
+    branches: [main]
+    paths:
+      - "benchmarks/sleeping/**"
+      - "utils/**"
+      - "tools/**"
+      - "runner/**"
+
+env:
+  GOOS: linux
+  GO111MODULE: on
+  PORT: 50051
+  PLATFORMS: linux/amd64,linux/arm64
+
+jobs:
+  build-and-push:
+    name: Build and push all images
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        service:
+          [
+            sleeping-go
+          ]
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v4
+        with:
+          lfs: "true"
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version: '1.21'
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build and push
+        env:
+            GOPRIVATE_KEY: ${{ secrets.XDT_REPO_ACCESS_KEY }}
+        uses: docker/build-push-action@v5
+        with:
+          push: true
+          file: benchmarks/sleeping/docker/Dockerfile
```

```suggestion
          file: ${{ matrix.file }}
```
Since you have different docker files you have to make this as a parameter.
Similar as you did for knative